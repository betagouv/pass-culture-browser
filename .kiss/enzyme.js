import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
