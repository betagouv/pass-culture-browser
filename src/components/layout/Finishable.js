import React from 'react'
import PropTypes from 'prop-types'

import { Icon } from 'pass-culture-shared'

const Finishable = ({ finished, children }) => {
  if (!finished) return children
  return (
    <div className="finishable">
      {children}
      <Icon svg="badge-termine" className="finish-icon" alt="TerminÃ©" />
    </div>
  )
}

Finishable.defaultProps = {
  finished: false,
}

Finishable.propTypes = {
  children: PropTypes.node.isRequired,
  finished: PropTypes.bool,
}

export default Finishable
