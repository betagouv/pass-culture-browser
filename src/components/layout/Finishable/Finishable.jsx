import PropTypes from 'prop-types'
import React from 'react'

import Icon from '../Icon/Icon'

const Finishable = ({ isNotBookable, children }) =>
  isNotBookable ? (
    <div className="finishable">
      {children}
      <Icon
        alt="RÃ©servation finie"
        className="finishable-ribbon-img"
        svg="badge-termine"
      />
    </div>
  ) : (
    children
  )

Finishable.defaultProps = {
  isNotBookable: false,
}

Finishable.propTypes = {
  children: PropTypes.node.isRequired,
  isNotBookable: PropTypes.bool,
}

export default Finishable
