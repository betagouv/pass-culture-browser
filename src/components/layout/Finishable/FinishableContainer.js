import { connect } from 'react-redux'
import { withRouter } from 'react-router-dom'
import { compose } from 'redux'

import Finishable from './Finishable'
import selectIsFinishedByRouterMatch from '../../../selectors/selectIsFinishedByRouterMatch'

const mapStateToProps = (state, ownProps) => {
  const { match } = ownProps
  const isFinished = selectIsFinishedByRouterMatch(state, match)
  return {
    isFinished,
  }
}

export default compose(
  withRouter,
  connect(mapStateToProps)
)(Finishable)
