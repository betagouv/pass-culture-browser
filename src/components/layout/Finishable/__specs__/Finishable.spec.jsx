import { shallow } from 'enzyme'
import React from 'react'

import Finishable from '../Finishable'

describe('src | components | layout | Finishable | Finishable', () => {
  let props

  beforeEach(() => {
    props = {
      children: <p>
        {'Cette offre n’est pas terminée'}
      </p>,
    }
  })

  describe('when the offer is bookable', () => {
    it('should not render the ribbon', () => {
      // given
      props.isNotBookable = false

      // when
      const wrapper = shallow(<Finishable {...props} />)

      // then
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when the offer is not bookable', () => {
    it('should render the ribbon', () => {
      // given
      props.isNotBookable = true

      // when
      const wrapper = shallow(<Finishable {...props} />)

      // then
      expect(wrapper).toMatchSnapshot()
    })
  })
})
