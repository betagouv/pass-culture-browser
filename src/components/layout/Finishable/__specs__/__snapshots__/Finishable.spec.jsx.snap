// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | layout | Finishable | Finishable when the offer is bookable should not render the ribbon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Finishable
    isNotBookable={false}
  >
    <p>
      Cette offre n’est pas terminée
    </p>
  </Finishable>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "Cette offre n’est pas terminée",
    },
    "ref": null,
    "rendered": "Cette offre n’est pas terminée",
    "type": "p",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "Cette offre n’est pas terminée",
      },
      "ref": null,
      "rendered": "Cette offre n’est pas terminée",
      "type": "p",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`src | components | layout | Finishable | Finishable when the offer is not bookable should render the ribbon 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Finishable
    isNotBookable={true}
  >
    <p>
      Cette offre n’est pas terminée
    </p>
  </Finishable>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <p>
          Cette offre n’est pas terminée
        </p>,
        <Icon
          alt="Réservation finie"
          className="finishable-ribbon-img"
          src={null}
          svg="badge-termine"
        />,
      ],
      "className": "finishable",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Cette offre n’est pas terminée",
        },
        "ref": null,
        "rendered": "Cette offre n’est pas terminée",
        "type": "p",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "alt": "Réservation finie",
          "className": "finishable-ribbon-img",
          "src": null,
          "svg": "badge-termine",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <p>
            Cette offre n’est pas terminée
          </p>,
          <Icon
            alt="Réservation finie"
            className="finishable-ribbon-img"
            src={null}
            svg="badge-termine"
          />,
        ],
        "className": "finishable",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Cette offre n’est pas terminée",
          },
          "ref": null,
          "rendered": "Cette offre n’est pas terminée",
          "type": "p",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "alt": "Réservation finie",
            "className": "finishable-ribbon-img",
            "src": null,
            "svg": "badge-termine",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
